// @pinpt/cards-react/src/util.ts
import { ENVIRONMENTS } from './types';

export const getCardEmbedUrl = (
	environment: ENVIRONMENTS,
	profile_id: string,
	editingEnabled: boolean,
	fullScreen: boolean,
	disableChrome: boolean,
	frameless: boolean,
) => {
	const qs = `&source=pinpoint-embed&editable=${editingEnabled ? 'true' : 'false'}&fullScreen=${fullScreen ? 'true' : 'false'}&chrome=${disableChrome ? 'false' : 'true'}&frameless=${frameless ? 'true' : 'false'}`;
	switch (environment) {
		case ENVIRONMENTS.LOCAL: return `https://cards.local.ppoint.io:3002/profile/${profile_id}?env=local${qs}`;
		case ENVIRONMENTS.EDGE: return `https://cards.pinpoint.com/profile/${profile_id}?env=edge${qs}`;
		case ENVIRONMENTS.STABLE:
		default: return `https://cards.pinpoint.com/profile/${profile_id}?env=stable${qs}`;
	}
};

export const getMockCardEmbedUrl = (
	avatar: string,
	firstName: string,
	lastName: string,
) => {
	const qs = `?avatar=${encodeURIComponent(avatar)}&first_name=${encodeURIComponent(firstName)}&last_name=${encodeURIComponent(lastName)}`;
	return `https://cards.pinpoint.com/profile/mock${qs}`;
};
